src/codegen/implicit-caller-location.md

```diff
@@ -0,0 +1,154 @@
+# Implicit Caller Location
+
+Approved in [RFC 2091], this feature enables the accurate reporting of caller location during panics
+initiated from functions like `Option::unwrap` and `Result::expect`. This feature adds the
+[`#[track_caller]`][attr-reference] attribute for functions, the [`caller_location`][intrinsic]
+intrinsic, and the stabilization-friendly [`core::panic::Location::caller`][wrapper] wrapper.
+
+## Motivating Example
+
+Take this small example program:
+
+```
+$ cat example.rs
+fn main() {
+    let foo: Option<()> = None;
+    foo.unwrap(); // this should produce a useful panic message!
+}
+```
+
+Prior to the implementation of RFC 2091, panics like this `unwrap()` printed a location in libcore:
+
+```
+$ rustc +1.39.0 example.rs && ./example
+thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/libcore/option.rs:378:21
+note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
+```
+
+After the implementation, we get a much more helpful message:
+
+```
+$ rustc +nightly-2020-01-20 example.rs && ./example
+thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', example.rs:3:5
+note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
+```
+
+## Background
+
+Broadly speaking, this feature's goal is to improve common Rust error messages without breaking
+stability guarantees, requiring modifications to end-user source, relying on platform-specific
+debug-info, or preventing user-defined types from having the same error-reporting benefits.
+
+Improving the output of these panics has been a goal of proposals since at least mid-2016 (see
+[non-viable alternatives] in the approved RFC for details). It took two more years until RFC 2091
+was approved, much of its [rationale] for this feature's design having been discovered through the
+discussion around several earlier proposals.
+
+The design in the original RFC limited itself to implementations that could be done inside the
+compiler at the time without significant refactoring. However in the year and a half between the
+approval of the RFC and the actual implementation work, a [revised design] was proposed and written
+up on the tracking issue. During the course of implementing that, it was also discovered that an
+implementation was possible without modifying the number of arguments in a function's MIR, which
+would simplify later stages and unlock use in traits.
+
+## The Attribute
+
+The `#[track_caller]` attribute is checked alongside other codegen attrs to ensure the function:
+
+* has `"Rust"` ABI (as opposed to `"C"`, etc...)
+* is not in an extern block
```

Comment:
We probably use "extern block" elsewhere, but I wish we could say "is not a foreign import". or "FFI import'

Eh, at the very least, maybe write `extern {...}` instead of "extern".